api/src/models/service.rs

```diff
@@ -36,8 +39,25 @@ pub struct Service {
     app_name: String,
     service_name: String,
     container_type: ContainerType,
-    container_id: String,
     base_url: Option<Url>,
+    endpoint: Option<ServiceEndpoint>,
+    web_host_meta: Option<WebHostMeta>,
+}
+
+#[derive(Clone, Debug)]
+struct ServiceEndpoint {
+    internal_addr: IpAddr,
+    exposed_port: u16,
+}
+
+impl ServiceEndpoint {
+    fn to_url(&self) -> Url {
+        Url::parse(&format!(
+            "http://{}:{}/",
```

Comment:
I guess SSL to the containers is not supported anyway? Not that it matters much when using virtual networks.

Reply:
Correct, currently PREvant assumes that there is an HTTP endpoint. But, if the container support HTTP redirects, [`reqwest` handles redirects:](https://docs.rs/reqwest/0.9.15/reqwest/#redirect-policy)

> By default, a Client will automatically handle HTTP redirects, detecting loops, and having a maximum redirect chain of 10 hops. To customize this behavior, a RedirectPolicy can used with a ClientBuilder.

In this case PREvant should be able to handle redirects to HTTPS. However, in the future we might have to handle self-signed certificates and stuff like this.